### DISCUSSION TITLE
Is there a way to get the page number for a Link src? (for a table of contents)

### DISCUSSION THREAD

**Discussion by fturmel** [wojtekmaj/react-pdf#1633] [Q&A]
I need to build a table of contents for a PDF that is somewhat unpredictable in structure. Is there a way to get the page number for a Link src target? There's no `render()` on `Link`, so it can't be done in the same way you can get the current page number.

(Fantastic library btw!)

**Comment by jiangxiaoqiang** [comment]
you can detect the current page number in the scroll event by using IntersectionObserver. and you can get the total page number on the Document onLoadSuccess.
```

const onDocumentLoadSuccess = (pdf: DocumentCallback) => {
        const { numPages } = pdf;
        setNumPages(numPages);
    }
```

**Reply by fturmel** [reply]
This is incredibly embarrassing, but I just realized I posted my question in the wrong "react-pdf" project... 🤦

I really appreciate your time and reply @jiangxiaoqiang
